File: game-server/src/main/java/net/dodian/uber/game/model/item/ItemContainer.java
package net.dodian.uber.game.model.item;

import java.util.ArrayList;

public class ItemContainer {

    private ArrayList<GameItem> items;

    public ItemContainer(int size) {
        this.items = new ArrayList<GameItem>();
        for (int i = 0; i < size; i++)
            items.add(new GameItem(-1, 0));
    }

    protected GameItem getSlot(int slot) {
        return items.get(slot);
    }

    protected void setSlot(int slot, GameItem item) {
        items.set(slot, item);
    }

}

File: game-server/bin/main/net/dodian/uber/game/model/item/ItemContainer.kt
package net.dodian.uber.game.model.item

import java.util.ArrayList

open class ItemContainer(size: Int) {
    private val items = ArrayList<GameItem>(size)

    init {
        for (index in 0 until size) {
            items.add(GameItem(-1, 0))
        }
    }

    protected fun getSlot(slot: Int): GameItem = items[slot]

    protected fun setSlot(slot: Int, item: GameItem) {
        items[slot] = item
    }
}
